test/active_record_doctor/detectors/incorrect_length_validation_test.rb

```diff
@@ -61,6 +61,40 @@ def test_no_validation_and_no_limit_is_ok
     refute_problems
   end
 
+  def test_array_inclusion_validation_with_shorter_values_and_limit_is_ok
+    Context.create_table(:users) do |t|
+      t.string :status, limit: 64
+    end.define_model do
+      validates :status, inclusion: { in: ["new", "vip"] }
+    end
+
+    refute_problems
+  end
+
+  def test_array_inclusion_validation_with_longer_values_and_limit_is_error
+    Context.create_table(:users) do |t|
+      t.string :status, limit: 64
+    end.define_model do
+      validates :status, inclusion: { in: ["new", "vip", "a" * 100] }
```

Comment:
I suggest using the smallest length that won't work to test the boundary condition - 65 in this case.